tests/reporting/test_metrics.py

```diff
@@ -0,0 +1,88 @@
+import datetime
+import json
+import os
+from unittest.mock import Mock, patch
+
+from fl4health.reporting.metrics import MetricsReporter
+
+
+@patch("fl4health.reporting.metrics.uuid")
+def test_metrics_reporter_init(mock_uuid: Mock) -> None:
+    test_uuid = "test uuid"
+    mock_uuid.uuid4.return_value = test_uuid
+
+    metrics_reporter = MetricsReporter()
+
+    assert metrics_reporter.run_id == test_uuid
+    assert metrics_reporter.metrics == {}
+
+
+def test_metrics_reporter_add_to_metrics() -> None:
+    test_data_1 = {"test data 1": 123}
+    test_data_2 = {"test data 2": 456}
+
+    metrics_reporter = MetricsReporter()
+
+    metrics_reporter.add_to_metrics(test_data_1)
+    assert metrics_reporter.metrics == test_data_1
+
+    metrics_reporter.add_to_metrics(test_data_2)
+    assert metrics_reporter.metrics == {**test_data_1, **test_data_2}
+
+
+def test_metrics_reporter_add_to_metrics_at_round() -> None:
+    test_data_1 = {"test data 1": 123}
+    test_data_2 = {"test data 2": 456}
+
+    metrics_reporter = MetricsReporter()
+
+    metrics_reporter.add_to_metrics_at_round(2, test_data_1)
+    assert metrics_reporter.metrics == {
+        "rounds": {
+            2: test_data_1,
+        },
+    }
+
+    metrics_reporter.add_to_metrics_at_round(4, test_data_1)
+    assert metrics_reporter.metrics == {
+        "rounds": {
+            2: test_data_1,
+            4: test_data_1,
+        },
+    }
+
+    metrics_reporter.add_to_metrics_at_round(2, test_data_2)
+    assert metrics_reporter.metrics == {
+        "rounds": {
+            2: {**test_data_1, **test_data_2},
+            4: test_data_1,
+        },
+    }
+
+
+def test_metrics_reporter_dump() -> None:
```

Comment:
pytest supports using a tmp folder that is created and cleaned up on the backend so you don't have to do it yourself. See `tests/checkpointing/test_save_load.py`. Just a suggestion.